### PR TITLE
add all local interfaces to node

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

File: models/node.go [hunk 1]
```diff
@@ -43,6 +43,12 @@ var seededRand *rand.Rand = rand.New(
 type NodeCheckin struct {
 	Version   string
 	Connected string
+	Ifaces    []Iface
+}
+
+type Iface struct {
```
Comment (0xdcarns) [inline models/node.go:49]: missing comment for public type

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: netclient/functions/join.go [hunk 2]
```diff
@@ -238,6 +238,14 @@ func JoinNetwork(cfg *config.ClientConfig, privateKey string) error {
 			logger.Log(1, "network:", cfg.Network, "error retrieving private address: ", err.Error())
 		}
 	}
+	if len(cfg.Node.Interfaces) == 0 {
+		ip, err := getInterfaces()
+		if err != nil {
+			logger.Log(0, "failed to retrive local interfaces", err.Error())
```
Comment (0xdcarns) [inline netclient/functions/join.go:244]: if there is an error, should we set interfaces to an empty array, in-case it's nil?

Comment (0xdcarns) [reply netclient/functions/join.go:244]: same to below file

Comment (mattkasun) [reply netclient/functions/join.go:244]: the only way this interfaces can get set to nil is if getInterfaces() returns nil,nil , ie no errors retrieving interfaces but no actual interfaces.  If someone is running netclient with no interfaces a panic isn't out of the question.

if the case of an error from getInterfaces(), continuing to use the previously generated array is less misleading than setting interfaces to an empty array.

**Review by 0xdcarns — APPROVED** [review]

**Review by alphadose — APPROVED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**